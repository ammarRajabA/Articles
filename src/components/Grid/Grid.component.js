import React, {Component} from 'react';
import { Button, CardImg, CardTitle, CardText,CardSubtitle, CardBody } from 'reactstrap';

import {stateToHTML} from 'draft-js-export-html';
import {convertFromRaw} from 'draft-js';

import './Grid.style.css';

class Grid extends Component{
	renderRowItems=(rowIndex)=>{
		var items=[]
		for (var i = 0; i < this.props.cols; i++) {
			let item=this.props.items[(rowIndex*this.props.cols)+i];
			if (item===undefined) return items;
			let contentState=convertFromRaw(JSON.parse(item.content))
			items.push(
					<span className="grid-card" key={i.toString()}>
		        		<CardImg top width="50%" height="50%" src={item.image} alt="Card image cap" />
		        		<CardBody>
		          			<div className="title">{item.title}</div>
		          			<div className="subtitle">{item.author}</div>
		          			<div className="content">
		          				<div dangerouslySetInnerHTML={{__html: stateToHTML(contentState).substr(0,100)+"..."}} />
		          			</div>
		          			<Button onClick={()=>this.props.handleSelect(item)}>Read more</Button>
		        		</CardBody>
		      		</span>
				)
		}
		return items;
	}
	renderRows=()=>{
		var rowsCount=Math.ceil(this.props.items.length/this.props.cols);
		var rows=[];
		for (var i = 0; i < rowsCount; i++) {
			rows.push(
					<span className="grid-card-deck" key={i.toString()}>
						{this.renderRowItems(i)}
					</span>
				)
		}
		return rows;
	}
	render(){
		return(
				<div className="grid-container">
					{this.renderRows()}
				</div>
			)
	}
}

export default Grid;