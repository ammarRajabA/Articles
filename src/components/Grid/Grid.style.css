.grid-container{
	display: flex;
	flex-direction: column;
	margin: 20px;
}
.grid-container .grid-card-deck{
	display: flex;
	flex-direction: row;
	justify-content: space-around;
	flex-wrap: wrap;
	width: 100%;
	margin-bottom: 20px;
}
.grid-card-deck .grid-card{
	width: 30%;
	border: 1px solid #E3E3E3;
}
.grid-card-deck .grid-card .title{
	font-weight: bold;
	font-size: 18px;
}
.grid-card-deck .grid-card .subtitle{
	font-size: 12px;
	font-style: italic;
}
.grid-card-deck .grid-card .content{
	margin-top: 20px;
	font-size: 14px;
}


@media (min-width: 320px) and (max-width: 767px) {
	.grid-card-deck .grid-card{
		width: 90%;
		border: 1px solid #E3E3E3;
	}
}