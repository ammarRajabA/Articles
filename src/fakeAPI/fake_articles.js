export default[
	{
		title:"first",
		content:JSON.stringify({"blocks":[{"key":"5viej","text":"This is my first article, it so cool I can color words ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":49,"length":5,"style":"color-rgb(243,121,52)"}],"entityRanges":[],"data":{}},{"key":"dh0hv","text":"I can also make bold italic or any other style.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":16,"length":5,"style":"BOLD"},{"offset":21,"length":7,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"blt1q","text":"I also can write math 25 =32","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":23,"length":2,"style":"SUPERSCRIPT"}],"entityRanges":[],"data":{}},{"key":"ddclt","text":"ðŸ“£ Attention here : My Github account Ammar Rajab ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":2,"length":48,"style":"color-rgb(209,72,65)"}],"entityRanges":[{"offset":38,"length":11,"key":0}],"data":{}}],"entityMap":{"0":{"type":"LINK","mutability":"MUTABLE","data":{"url":"https://github.com/ammarrajaba","targetOption":"_blank"}}}}),
		image:"https://picsum.photos/id/258/200/300",
		author:"Ammar Rajab",
		id:"13245"
	},
	{
		title:"Second",
		content:JSON.stringify({"blocks":[{"key":"35tp4","text":"Another article by me,","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"a1v72","text":"It shows how we align text","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"cifh6","text":"Left","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"b5gsb","text":"Middle","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"center"}},{"key":"eopjr","text":"and right","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"right"}},{"key":"1h6bs","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"d3mrc","text":"and change font","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":0,"length":15,"style":"fontfamily-Impact"}],"entityRanges":[],"data":{}}],"entityMap":{}}),
		image:"https://picsum.photos/id/232/200/300",
		author:"Ammar Rajab",
		id:"15555"
	},
	{
		title:"Third",
		content:JSON.stringify({"blocks":[{"key":"5viej","text":"This is my first article, it so cool I can color words ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":49,"length":5,"style":"color-rgb(243,121,52)"}],"entityRanges":[],"data":{}},{"key":"dh0hv","text":"I can also make bold italic or any other style.","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":16,"length":5,"style":"BOLD"},{"offset":21,"length":7,"style":"ITALIC"}],"entityRanges":[],"data":{}},{"key":"blt1q","text":"I also can write math 25 =32","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":23,"length":2,"style":"SUPERSCRIPT"}],"entityRanges":[],"data":{}},{"key":"ddclt","text":"ðŸ“£ Attention here : My Github account Ammar Rajab ","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":2,"length":48,"style":"color-rgb(209,72,65)"}],"entityRanges":[{"offset":38,"length":11,"key":0}],"data":{}}],"entityMap":{"0":{"type":"LINK","mutability":"MUTABLE","data":{"url":"https://github.com/ammarrajaba","targetOption":"_blank"}}}}),
		image:"https://picsum.photos/id/258/200/300",
		author:"Ammar Rajab",
		id:"13445"
	},
	{
		title:"Forth",
		content:JSON.stringify({"blocks":[{"key":"35tp4","text":"Another article by me,","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"a1v72","text":"It shows how we align text","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"cifh6","text":"Left","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"b5gsb","text":"Middle","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"center"}},{"key":"eopjr","text":"and right","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{"text-align":"right"}},{"key":"1h6bs","text":"","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}},{"key":"d3mrc","text":"and change font","type":"unstyled","depth":0,"inlineStyleRanges":[{"offset":0,"length":15,"style":"fontfamily-Impact"}],"entityRanges":[],"data":{}}],"entityMap":{}}),
		image:"https://picsum.photos/id/232/200/300",
		author:"Ammar Rajab",
		id:"15005"
	}
]
