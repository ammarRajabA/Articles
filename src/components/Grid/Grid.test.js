import React from 'react'
import { shallow ,mount } from 'enzyme';

import Grid from './Grid.component';
import fake_articles from '../../fakeAPI/fake_articles'

describe('Grid component',function () {
	it('match snapshot',function () {
		const component = mount(<Grid items={fake_articles} cols={3}/>);
		expect(component).toMatchSnapshot();
	})
})