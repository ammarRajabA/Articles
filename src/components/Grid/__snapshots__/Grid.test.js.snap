// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component match snapshot 1`] = `
<Grid
  cols={3}
  items={
    Array [
      Object {
        "author": "Ammar Rajab",
        "content": "{\\"blocks\\":[{\\"key\\":\\"5viej\\",\\"text\\":\\"This is my first article, it so cool I can color words \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":49,\\"length\\":5,\\"style\\":\\"color-rgb(243,121,52)\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"dh0hv\\",\\"text\\":\\"I can also make bold italic or any other style.\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":16,\\"length\\":5,\\"style\\":\\"BOLD\\"},{\\"offset\\":21,\\"length\\":7,\\"style\\":\\"ITALIC\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"blt1q\\",\\"text\\":\\"I also can write math 25 =32\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":23,\\"length\\":2,\\"style\\":\\"SUPERSCRIPT\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"ddclt\\",\\"text\\":\\"ðŸ“£ Attention here : My Github account Ammar Rajab \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":2,\\"length\\":48,\\"style\\":\\"color-rgb(209,72,65)\\"}],\\"entityRanges\\":[{\\"offset\\":38,\\"length\\":11,\\"key\\":0}],\\"data\\":{}}],\\"entityMap\\":{\\"0\\":{\\"type\\":\\"LINK\\",\\"mutability\\":\\"MUTABLE\\",\\"data\\":{\\"url\\":\\"https://github.com/ammarrajaba\\",\\"targetOption\\":\\"_blank\\"}}}}",
        "id": "13245",
        "image": "https://picsum.photos/id/258/200/300",
        "title": "first",
      },
      Object {
        "author": "Ammar Rajab",
        "content": "{\\"blocks\\":[{\\"key\\":\\"35tp4\\",\\"text\\":\\"Another article by me,\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"a1v72\\",\\"text\\":\\"It shows how we align text\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"cifh6\\",\\"text\\":\\"Left\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"b5gsb\\",\\"text\\":\\"Middle\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"center\\"}},{\\"key\\":\\"eopjr\\",\\"text\\":\\"and right\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"right\\"}},{\\"key\\":\\"1h6bs\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"d3mrc\\",\\"text\\":\\"and change font\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":0,\\"length\\":15,\\"style\\":\\"fontfamily-Impact\\"}],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
        "id": "15555",
        "image": "https://picsum.photos/id/232/200/300",
        "title": "Second",
      },
      Object {
        "author": "Ammar Rajab",
        "content": "{\\"blocks\\":[{\\"key\\":\\"5viej\\",\\"text\\":\\"This is my first article, it so cool I can color words \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":49,\\"length\\":5,\\"style\\":\\"color-rgb(243,121,52)\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"dh0hv\\",\\"text\\":\\"I can also make bold italic or any other style.\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":16,\\"length\\":5,\\"style\\":\\"BOLD\\"},{\\"offset\\":21,\\"length\\":7,\\"style\\":\\"ITALIC\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"blt1q\\",\\"text\\":\\"I also can write math 25 =32\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":23,\\"length\\":2,\\"style\\":\\"SUPERSCRIPT\\"}],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"ddclt\\",\\"text\\":\\"ðŸ“£ Attention here : My Github account Ammar Rajab \\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":2,\\"length\\":48,\\"style\\":\\"color-rgb(209,72,65)\\"}],\\"entityRanges\\":[{\\"offset\\":38,\\"length\\":11,\\"key\\":0}],\\"data\\":{}}],\\"entityMap\\":{\\"0\\":{\\"type\\":\\"LINK\\",\\"mutability\\":\\"MUTABLE\\",\\"data\\":{\\"url\\":\\"https://github.com/ammarrajaba\\",\\"targetOption\\":\\"_blank\\"}}}}",
        "id": "13445",
        "image": "https://picsum.photos/id/258/200/300",
        "title": "Third",
      },
      Object {
        "author": "Ammar Rajab",
        "content": "{\\"blocks\\":[{\\"key\\":\\"35tp4\\",\\"text\\":\\"Another article by me,\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"a1v72\\",\\"text\\":\\"It shows how we align text\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"cifh6\\",\\"text\\":\\"Left\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"b5gsb\\",\\"text\\":\\"Middle\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"center\\"}},{\\"key\\":\\"eopjr\\",\\"text\\":\\"and right\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{\\"text-align\\":\\"right\\"}},{\\"key\\":\\"1h6bs\\",\\"text\\":\\"\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[],\\"entityRanges\\":[],\\"data\\":{}},{\\"key\\":\\"d3mrc\\",\\"text\\":\\"and change font\\",\\"type\\":\\"unstyled\\",\\"depth\\":0,\\"inlineStyleRanges\\":[{\\"offset\\":0,\\"length\\":15,\\"style\\":\\"fontfamily-Impact\\"}],\\"entityRanges\\":[],\\"data\\":{}}],\\"entityMap\\":{}}",
        "id": "15005",
        "image": "https://picsum.photos/id/232/200/300",
        "title": "Forth",
      },
    ]
  }
>
  <div
    className="grid-container"
  >
    <span
      className="grid-card-deck"
      key="0"
    >
      <span
        className="grid-card"
        key="0"
      >
        <CardImg
          alt="Card image cap"
          height="50%"
          src="https://picsum.photos/id/258/200/300"
          tag="img"
          top={true}
          width="50%"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            height="50%"
            src="https://picsum.photos/id/258/200/300"
            width="50%"
          />
        </CardImg>
        <CardBody
          tag="div"
        >
          <div
            className="card-body"
          >
            <div
              className="title"
            >
              first
            </div>
            <div
              className="subtitle"
            >
              Ammar Rajab
            </div>
            <div
              className="content"
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>This is my first article, it so cool I can color words&nbsp;</p>
<p>I can also make <strong>bold ...",
                  }
                }
              />
            </div>
            <Button
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Read more
              </button>
            </Button>
          </div>
        </CardBody>
      </span>
      <span
        className="grid-card"
        key="1"
      >
        <CardImg
          alt="Card image cap"
          height="50%"
          src="https://picsum.photos/id/232/200/300"
          tag="img"
          top={true}
          width="50%"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            height="50%"
            src="https://picsum.photos/id/232/200/300"
            width="50%"
          />
        </CardImg>
        <CardBody
          tag="div"
        >
          <div
            className="card-body"
          >
            <div
              className="title"
            >
              Second
            </div>
            <div
              className="subtitle"
            >
              Ammar Rajab
            </div>
            <div
              className="content"
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Another article by me,</p>
<p>It shows how we align text</p>
<p>Left</p>
<p>Middle</p>
<p>and rig...",
                  }
                }
              />
            </div>
            <Button
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Read more
              </button>
            </Button>
          </div>
        </CardBody>
      </span>
      <span
        className="grid-card"
        key="2"
      >
        <CardImg
          alt="Card image cap"
          height="50%"
          src="https://picsum.photos/id/258/200/300"
          tag="img"
          top={true}
          width="50%"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            height="50%"
            src="https://picsum.photos/id/258/200/300"
            width="50%"
          />
        </CardImg>
        <CardBody
          tag="div"
        >
          <div
            className="card-body"
          >
            <div
              className="title"
            >
              Third
            </div>
            <div
              className="subtitle"
            >
              Ammar Rajab
            </div>
            <div
              className="content"
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>This is my first article, it so cool I can color words&nbsp;</p>
<p>I can also make <strong>bold ...",
                  }
                }
              />
            </div>
            <Button
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Read more
              </button>
            </Button>
          </div>
        </CardBody>
      </span>
    </span>
    <span
      className="grid-card-deck"
      key="1"
    >
      <span
        className="grid-card"
        key="0"
      >
        <CardImg
          alt="Card image cap"
          height="50%"
          src="https://picsum.photos/id/232/200/300"
          tag="img"
          top={true}
          width="50%"
        >
          <img
            alt="Card image cap"
            className="card-img-top"
            height="50%"
            src="https://picsum.photos/id/232/200/300"
            width="50%"
          />
        </CardImg>
        <CardBody
          tag="div"
        >
          <div
            className="card-body"
          >
            <div
              className="title"
            >
              Forth
            </div>
            <div
              className="subtitle"
            >
              Ammar Rajab
            </div>
            <div
              className="content"
            >
              <div
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<p>Another article by me,</p>
<p>It shows how we align text</p>
<p>Left</p>
<p>Middle</p>
<p>and rig...",
                  }
                }
              />
            </div>
            <Button
              color="secondary"
              onClick={[Function]}
              tag="button"
            >
              <button
                aria-label={null}
                className="btn btn-secondary"
                onClick={[Function]}
                type="button"
              >
                Read more
              </button>
            </Button>
          </div>
        </CardBody>
      </span>
    </span>
  </div>
</Grid>
`;
